### PR TITLE
NEML integration

### PR DESCRIPTION
**Pull request by bwspenc**
ref #59

### CODE REVIEW

File: include/materials/NEMLStress.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#ifndef NEMLSTRESS_H
```
Comment (dschwen) [inline include/materials/NEMLStress.h]: Header?

File: include/materials/NEMLThermalExpansionEigenstrain.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#ifndef COMPUTETHERMALEXPANSIONEIGENSTRAINNEML_H
+#define COMPUTETHERMALEXPANSIONEIGENSTRAINNEML_H
```
Comment (dschwen) [inline include/materials/NEMLThermalExpansionEigenstrain.h]: rename

**Comment by bwspenc** [comment]
@reverenedbedford FYI, I took the commit from your PR and built onto it in this PR. Once we get this all sorted out, we'll just want to close your original PR.

I talked this over with @dschwen, and I think we'll make a change to ComputeStressBase in MOOSE to make it so that it doesn't require an elasticity tensor. Once we get that change pushed through, I'll go back and remove that stuff from the NEML stress calculator.

**Comment by bwspenc** [comment]
@reverendbedford I'm having a heck of a time getting this to compile on all of our platforms due to the libxml dependency. It compiles and runs fine on my mac in opt and debug mode. I got the libxml++ dev package installed on the linux box that we do our testing on, and I thought I had it working, but now it's failing because of these undefined references to libxml++ functions:

Linking Executable /opt/civet/build_0/blackbear/blackbear-dbg...
/opt/civet/build_0/blackbear/contrib/neml/libneml-dbg.so: undefined reference to `xmlpp::Element::get_attributes() const'
/opt/civet/build_0/blackbear/contrib/neml/libneml-dbg.so: undefined reference to `xmlpp::Node::get_children(Glib::ustring const&) const'
collect2: error: ld returned 1 exit status

I've got an idea to get around these issues with libxml++. We're already dependent on the boost library, and that is one of the libraries that we have good support for in our packages. It looks like the boost property_tree can parse xml. If we could swap libxml++ for that, that would make it a lot easier to support neml with our regular packages. I'm going to attempt to modify neml to use property_tree. If you have any objections to that before I go too far down that path, let me know.

**Comment by reverendbedford** [comment]
I have no objection at all.  In fact if it works I would appreciate a pull request onto NEML proper, as I would really like to eliminate all the libxml++ dependencies.  They make compiling on windows even more of a pain than it normally is.

The issue with libxml is likely due to an API change they made awhile ago.  I have compiler macros that are supposed to switch between the different APIs (there were not many changes) but it could be I missed something (like that form of get_children?)

**Comment by bwspenc** [comment]
OK, good to know.  If we get it working, we'll definitely submit a pull request to NEML. We will have NEML be a submodule to BlackBear, so that we would always be pointing to the main repository, and periodically update the version of NEML after we verify that the new versions still pass our tests.

**Comment by bwspenc** [comment]
@dschwen At long last, I finally have this passing!  We got rid of the libxml++ dependency. If you could please take another look at this, that would be great.

**Comment by bwspenc** [comment]
FYI @reverendbedford I think this is ready to go.

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[submodule "contrib/neml"]
+	path = contrib/neml
+	url = ../../Argonne-National-Laboratory/neml.git
```
[the remark this answers is not in the document]
Comment (bwspenc) [reply .gitmodules]: Doing it this way gets it through either ssh or https, using whatever protocol you used to clone the repository. I think this is how we grab the libmesh submodule in moose.

[the remark this answers is not in the document]
Comment (bwspenc) [reply .gitmodules]: It works because they’re both hosted on GitHub.

[the remark this answers is not in the document]
Comment (dschwen) [reply .gitmodules]: Oh, cool. I guess I learned something new: Submodule urls can be relative to the superproject’s default remote repository (the remote of the remote-tracking branch of the current branch!)

https://git-scm.com/docs/git-submodule

**Pull request merged**